UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentCallBackService.cs

```diff
@@ -0,0 +1,95 @@
+using UKHO.ExchangeSetService.Common.Models.Response;
+using UKHO.ExchangeSetService.Common.Models.SalesCatalogue;
+using UKHO.ExchangeSetService.FulfilmentService.Configuration;
+
+namespace UKHO.ExchangeSetService.FulfilmentService.Services
+{
+    public class FulfilmentCallBackService : IFulfilmentCallBackService
+    {
+        private readonly IOptions<EssCallBackConfiguration> essCallBackConfiguration;
+        private readonly ICallBackClient callBackClient;
+        private readonly IOptions<FileShareServiceConfiguration> fileShareServiceConfig;
+        private readonly ILogger<FulfilmentCallBackService> logger;
+
+        public FulfilmentCallBackService(IOptions<EssCallBackConfiguration> essCallBackConfiguration,
+                                         ICallBackClient callBackClient,
+                                         IOptions<FileShareServiceConfiguration> fileShareServiceConfig,
+                                         ILogger<FulfilmentCallBackService> logger)
+        {
+            this.essCallBackConfiguration = essCallBackConfiguration;
+            this.callBackClient = callBackClient;
+            this.fileShareServiceConfig = fileShareServiceConfig;
+            this.logger = logger;
+        }
+
+        public async Task<bool> SendCallBackResponse(SalesCatalogueProductResponse salesCatalogueProductResponse, SalesCatalogueServiceResponseQueueMessage scsResponseQueueMessage)
+        {
+            if (!string.IsNullOrWhiteSpace(scsResponseQueueMessage.CallbackUri))
+            {
+                ExchangeSetResponse exchangeSetResponse = new ExchangeSetResponse()
+                {
+                    Links = new Links()
+                    {
+                        ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = $"{fileShareServiceConfig.Value.BaseUrl}/batch/{scsResponseQueueMessage.BatchId}" },
+                        ExchangeSetFileUri = new LinkSetFileUri { Href = $"{fileShareServiceConfig.Value.BaseUrl}/batch/{scsResponseQueueMessage.BatchId}/files/{fileShareServiceConfig.Value.ExchangeSetFileName}" }
+                    },
+                    ExchangeSetUrlExpiryDateTime = Convert.ToDateTime(scsResponseQueueMessage.ExchangeSetUrlExpiryDate).ToUniversalTime(),
+                    RequestedProductCount = salesCatalogueProductResponse.ProductCounts.RequestedProductCount.Value,
+                    ExchangeSetCellCount = salesCatalogueProductResponse.ProductCounts.ReturnedProductCount.Value,
+                    RequestedProductsAlreadyUpToDateCount = salesCatalogueProductResponse.ProductCounts.RequestedProductsAlreadyUpToDateCount.Value,
+                    RequestedProductsNotInExchangeSet = GetRequestedProductsNotInExchangeSet(salesCatalogueProductResponse)
+                };
+
+                CallBackResponse callBackResponse = new CallBackResponse()
+                {
+                    Specversion = essCallBackConfiguration.Value.SpecVersion,
```

Comment:
Should this be SpecVersion?

Reply:
As per the suggestion, i have made changes should this be SpecVersion.